### PR TITLE
Implement `Check*Exists` APIs in transient mode

### PR DESCRIPTION
**Pull request by Y-Sindo**

### CODE REVIEW

File: src/Microsoft.Azure.SignalR.Management/RestHubLifetimeManager.cs [hunk 1]
```diff
@@ -263,6 +263,78 @@ private static void ValidateUserIdAndGroupName(string userId, string groupName)
...
+            {
+                switch (response.StatusCode)
+                {
+                    case HttpStatusCode.OK:
+                        exists = true;
+                        return true;
+                    case HttpStatusCode.NotFound:
+                        return true;
+                    default:
+                        return false;
```
Comment (vwxyzh) [inline src/Microsoft.Azure.SignalR.Management/RestHubLifetimeManager.cs:332]: what about service return 500?

File: src/Microsoft.Azure.SignalR.Management/RestHubLifetimeManager.cs [hunk 2]
```diff
@@ -263,6 +263,78 @@ private static void ValidateUserIdAndGroupName(string userId, string groupName)
...
```
Comment (vwxyzh) [inline src/Microsoft.Azure.SignalR.Management/RestHubLifetimeManager.cs:308]: what about returning 500?

File: src/Microsoft.Azure.SignalR.Management/RestHubLifetimeManager.cs (hunk 1 above)
Comment (Y-Sindo) [reply src/Microsoft.Azure.SignalR.Management/RestHubLifetimeManager.cs:332]: An `AzureSignalRRuntimeException` will be thrown.
https://github.com/Azure/azure-signalr/blob/f0424627c6be163fe5e3077a190a3a8e8979c058/src/Microsoft.Azure.SignalR.Common/Utilities/RestClient.cs#L93

File: src/Microsoft.Azure.SignalR.Management/HubContext/ClientManager.cs [hunk 3]
```diff
@@ -18,5 +18,29 @@ internal abstract class ClientManager
...
         /// <remarks>To get the <paramref name="reason"/> from connection closed event, client should set <see cref="NegotiationOptions.EnableDetailedErrors"/> during negotiation.</remarks>
         public abstract Task CloseConnectionAsync(string connectionId, string reason = null, CancellationToken cancellationToken = default);
+
+        /// <summary>
+        /// Check if a connection exists asynchronously.
+        /// </summary>
+        /// <param name="connectionId">The connection to check</param>
+        /// <param name="cancellationToken"></param>
+        /// <returns>The created <see cref="System.Threading.Tasks.Task{TResult}">Task</see> that represents the asynchronous operation. True if the connection exists, otherwise false.</returns>
+        public abstract Task<bool> CheckIfConnectionExistsAsync(string connectionId, CancellationToken cancellationToken = default);
```
Comment (vicancy) [inline src/Microsoft.Azure.SignalR.Management/HubContext/ClientManager.cs:28]: `ConnectionExistsAsync` is enough?

File: src/Microsoft.Azure.SignalR.Management/RestHubLifetimeManager.cs [hunk 4]
```diff
@@ -263,6 +263,78 @@ private static void ValidateUserIdAndGroupName(string userId, string groupName)
...
+            var api = await _restApiProvider.GetCheckConnectionExistsEndpointAsync(_appName, _hubName, connectionId);
+            await _restClient.SendAsync(api, HttpMethod.Head, _productInfo, handleExpectedResponse: response =>
+            {
+                switch (response.StatusCode)
+                {
+                    case HttpStatusCode.OK:
+                        exists = true;
+                        return true;
+                    case HttpStatusCode.NotFound:
+                        return true;
```
Comment (vicancy) [inline src/Microsoft.Azure.SignalR.Management/RestHubLifetimeManager.cs:282]: return false?

Comment (Y-Sindo) [reply src/Microsoft.Azure.SignalR.Management/RestHubLifetimeManager.cs:282]: The return value indicates whether to throw an exception. The variable used to indicate existing is `exists`.

Comment (Y-Sindo) [reply src/Microsoft.Azure.SignalR.Management/RestHubLifetimeManager.cs:282]: Intend to refactor `RestClient` later

Comment (vicancy) [reply src/Microsoft.Azure.SignalR.Management/RestHubLifetimeManager.cs:282]: I see

File: src/Microsoft.Azure.SignalR.Management/HubContext/ClientManager.cs [hunk 5]
```diff
@@ -18,5 +18,29 @@ internal abstract class ClientManager
         /// <returns>The created <see cref="System.Threading.Tasks.Task{TResult}">Task</see> that represents the asynchronous operation.</returns>
         /// <remarks>To get the <paramref name="reason"/> from connection closed event, client should set <see cref="NegotiationOptions.EnableDetailedErrors"/> during negotiation.</remarks>
         public abstract Task CloseConnectionAsync(string connectionId, string reason = null, CancellationToken cancellationToken = default);
+
+        /// <summary>
+        /// Check if a connection exists asynchronously.
+        /// </summary>
+        /// <param name="connectionId">The connection to check</param>
+        /// <param name="cancellationToken"></param>
+        /// <returns>The created <see cref="System.Threading.Tasks.Task{TResult}">Task</see> that represents the asynchronous operation. True if the connection exists, otherwise false.</returns>
```
Comment (wanlwanl) [inline src/Microsoft.Azure.SignalR.Management/HubContext/ClientManager.cs:27]: Task{TResult} -> Task<bool>

File: src/Microsoft.Azure.SignalR.Management/HubContext/ClientManager.cs [hunk 6]
```diff
@@ -18,5 +18,29 @@ internal abstract class ClientManager
         /// <returns>The created <see cref="System.Threading.Tasks.Task{TResult}">Task</see> that represents the asynchronous operation.</returns>
         /// <remarks>To get the <paramref name="reason"/> from connection closed event, client should set <see cref="NegotiationOptions.EnableDetailedErrors"/> during negotiation.</remarks>
         public abstract Task CloseConnectionAsync(string connectionId, string reason = null, CancellationToken cancellationToken = default);
+
+        /// <summary>
+        /// Check if a connection exists asynchronously.
+        /// </summary>
+        /// <param name="connectionId">The connection to check</param>
+        /// <param name="cancellationToken"></param>
```
Comment (wanlwanl) [inline src/Microsoft.Azure.SignalR.Management/HubContext/ClientManager.cs:26]: nit: add comment

Comment (wanlwanl) [reply src/Microsoft.Azure.SignalR.Management/HubContext/ClientManager.cs:26]: also apply to other functions

**Review by vicancy — APPROVED** [review]

**Comment by wanlwanl** [comment]
why delete the parameters but keep other comments like return value?
api doc will get empty in paramters section this way.
@vicancy @Y-Sindo

**Pull request merged**